extensions/ql-vscode/src/variant-analysis/store/repo-states-to-data-mapper.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  VariantAnalysisScannedRepositoryDownloadStatus,
+  VariantAnalysisScannedRepositoryState,
+} from "../shared/variant-analysis";
+import {
+  VariantAnalysisScannedRepositoryDownloadData,
+  VariantAnalysisScannedRepositoryStateData,
+} from "./repo-states-data-types";
+
+export function mapRepoStateToData(
+  repoState: VariantAnalysisScannedRepositoryState,
+): VariantAnalysisScannedRepositoryStateData {
+  return {
+    repositoryId: repoState.repositoryId,
+    downloadStatus: processDownloadStatus(repoState.downloadStatus),
+    downloadPercentage: repoState.downloadPercentage,
+  };
+}
+
+function processDownloadStatus(
+  downloadedStatus: VariantAnalysisScannedRepositoryDownloadStatus,
+) {
+  switch (downloadedStatus) {
+    case VariantAnalysisScannedRepositoryDownloadStatus.Pending:
+      return VariantAnalysisScannedRepositoryDownloadData.Pending;
+    case VariantAnalysisScannedRepositoryDownloadStatus.InProgress:
+      return VariantAnalysisScannedRepositoryDownloadData.InProgress;
+    case VariantAnalysisScannedRepositoryDownloadStatus.Succeeded:
+      return VariantAnalysisScannedRepositoryDownloadData.Succeeded;
+    case VariantAnalysisScannedRepositoryDownloadStatus.Failed:
+      return VariantAnalysisScannedRepositoryDownloadData.Failed;
+    default:
+      return VariantAnalysisScannedRepositoryDownloadData.Pending;
```

Comment:
```suggestion
      assertNever(downloadedStatus);
```

This is another good case for using `assertNever`.  When mapping from domain to data models, we don't have to worry about badly typed data and having to handle every possibility, so we can say this case will never happen and the type system will enforce it.